frontend/packages/ceph-storage-plugin/src/components/block-pool/body.tsx

```diff
@@ -0,0 +1,269 @@
+      </EmptyState>
+    </>
+  );
+};
+
+export type BlockPoolStatusProps = {
+  status: string;
+  name?: string;
+  error?: string;
+};
+
+export const BlockPoolBody = (props: BlockPoolBodyPros) => {
+  const { cephCluster, state, dispatch, showPoolStatus } = props;
+  const { t } = useTranslation();
+
+  const isPoolManagementSupported = useFlag(GUARDED_FEATURES.OCS_POOL_MANAGEMENT);
+  const [storageCluster, storageClusterLoaded, storageClusterLoadError] = useK8sGet<
+    ListKind<StorageClusterKind>
+  >(OCSServiceModel, null, CEPH_STORAGE_NAMESPACE);
+
+  const [isReplicaOpen, setReplicaOpen] = React.useState(false);
+  const [isVolumeTypeOpen, setVolumeTypeOpen] = React.useState(false);
+  // Check storage cluster is in ready state
+  const isClusterReady: boolean = cephCluster?.status?.phase === POOL_STATE.READY;
+
+  // Check storage cluster is arbiter
+  React.useEffect(() => {
+    const isArbiterCluster: boolean = checkArbiterCluster(storageCluster?.items[0]);
+    dispatch({ type: BlockPoolActionType.SET_POOL_ARBITER, payload: isArbiterCluster });
+    if (isArbiterCluster) {
+      dispatch({ type: BlockPoolActionType.SET_POOL_REPLICA_SIZE, payload: '4' });
+    }
+  }, [
+    storageCluster,
+    storageClusterLoaded,
+    storageClusterLoadError,
+    state.isArbiterCluster,
+    dispatch,
+  ]);
+
+  React.useEffect(() => {
+    dispatch({
+      type: BlockPoolActionType.SET_ERROR_MESSAGE,
+      payload: !isClusterReady
+        ? t(
+            'ceph-storage-plugin~The OpenShift Storage cluster is not available. Try again after the storage cluster is ready to use.',
+          )
+        : '',
+    });
+  }, [dispatch, isClusterReady, t]);
+
+  const replicaList: string[] = _.keys(OCS_DEVICE_REPLICA).filter(
+    (replica: string) =>
+      (state.isArbiterCluster && replica === '4') || (!state.isArbiterCluster && replica !== '4'),
+  );
+
+  const replicaDropdownItems = replicaList.map((replica) => (
+    <DropdownItem
+      key={`replica-${OCS_DEVICE_REPLICA[replica]}`}
+      component="button"
+      id={replica}
+      data-test-id={replica}
+      onClick={(e) =>
+        dispatch({ type: BlockPoolActionType.SET_POOL_REPLICA_SIZE, payload: e.currentTarget.id })
+      }
+    >
+      {t('ceph-storage-plugin~{{replica}} Replication', { replica: OCS_DEVICE_REPLICA[replica] })}
+    </DropdownItem>
+  ));
+
+  // Volume Type
+  let availableDeviceClasses = [];
+  const deviceClasses = cephCluster?.status?.storage?.deviceClasses ?? [];
+  const setVolumeType = (volumeType: string) =>
+    dispatch({ type: BlockPoolActionType.SET_POOL_VOLUME_TYPE, payload: volumeType });
+
+  if (deviceClasses.length && isPoolManagementSupported) {
+    if (state.volumeType === '') {
+      // Set default value
+      const ssdDeviceClass = deviceClasses.find((deviceClass) => deviceClass.name === 'ssd') || {};
+      Object.keys(ssdDeviceClass).length
+        ? setVolumeType('ssd')
+        : setVolumeType(deviceClasses[0].name);
+    }
```

Comment:
Place this inside useEffect.
Otherwise it may cause re-renders

Reply:
If there is any change in device classes filed of ceph as well, then to it wont update.
It might not be the case now, but this sort of update leaves space for bugs in general.

Reply:
ack